app/javascript/controllers/form_controller.js

```diff
@@ -4,6 +4,7 @@ import { Controller } from "@hotwired/stimulus"
 // Connects to data-controller="form"
 export default class extends Controller {
   connect() {
+    this.element.addEventListener("keydown", (e) => e?.key === "Enter" && this.submit(e))
```

Comment:
i think the problem lies in the way PhlexUI render components (templates cloning) in UI.

To add explanations, I believe we should be able to submit a form with an `Enter` press natively without further implementations. The way Phlex UI is rendering the component in the DOM makes it as the button is not in the same scope as the form to submit it on `Enter` press.


If you try to remove the Form button outside the `<%= render PhlexUI::Dialog::Footer.new(class: "mt-4 pt-4 border-t border-gray-100") do %>`

you'll notice that `Enter` press will start to work.

I'd suggest to keep the normal `form + button/input 'submit'` rather than adding a custom `keydown` listener